pkg/rtc/dynacastmanager.go

```diff
@@ -0,0 +1,234 @@
+package rtc
+
+import (
+	"sync"
+	"time"
+
+	"github.com/bep/debounce"
+	"github.com/livekit/protocol/livekit"
+	"github.com/livekit/protocol/logger"
+
+	"github.com/livekit/livekit-server/pkg/rtc/types"
+	"github.com/livekit/livekit-server/pkg/utils"
+)
+
+type DynacastManagerParams struct {
+	DynacastPauseDelay time.Duration
+	Logger             logger.Logger
+}
+
+type DynacastManager struct {
```

Comment:
A lot of this PR is just refactoring. There is a `DynacastQuality` (another new module 👇 ) module per receiver in a multi-codec receiver. This module `DynacastManager` aggregates all the codecs of a receiver. Splitting these out into smaller modules as readability in `MediaTrackSubscriptions` was getting hard. And also, for relay track, these are maintained by relay manager and hence made sense to split them out.